application/modules/core/helpers/ContentBlockHelper.php

```diff
@@ -0,0 +1,197 @@
+        if (!empty($matches)) {
+            foreach ($matches[0] as $k => $rawChunk) {
+                $chunkData = static::sanitizeChunk($rawChunk);
+                $cacheChunkKey = $chunkData['key'].$content_key;
+                $replacement = Yii::$app->cache->get($cacheChunkKey);
+                if ($replacement === false) {
+                    $chunk = self::fetchChunkByKey($chunkData['key']);
+                    $replacement = static::compileChunk($chunk, $chunkData);
+                    if (null !== $chunk ) {
+                        Yii::$app->cache->set(
+                            $cacheChunkKey,
+                            $replacement,
+                            84600,
+                            $dependency
+                        );
+                    }
+                }
+                $content = str_replace($matches[0][$k], $replacement, $content);
+            }
+        }
+        return $content;
+    }
+
+    /**
+     * Extracting chunk data from chunk call
+     * @param $rawChunk
+     * @return array
+     */
+    private static function sanitizeChunk($rawChunk) {
+        $chunk = [];
+        preg_match('%\$([^\s\]]+)[\s\]]%', $rawChunk, $keyMatches);
+        $chunk['key'] = $keyMatches[1];
+        $expression = "#\s*(?P<paramName>[\\w\\d]*)=(('(?P<escapedValue>.*[^\\\\])')|(?P<unescapedValue>.*))(\\|(('(?P<escapedDefault>.*[^\\\\])')|(?P<unescapedDefault>.*)))?[\\]\\s]#uUi";
+        preg_match_all($expression, $rawChunk, $matches);
+        foreach ($matches['paramName'] as $key => $paramName) {
+            if (!empty($matches['escapedValue'][$key])) {
+                $chunk[$paramName] = strval($matches['escapedValue'][$key]);
+            }
+            if (!empty($matches['unescapedValue'][$key])) {
+                $chunk[$paramName] = floatval($matches['unescapedValue'][$key]);
+            }
+            if (!empty($matches['escapedDefault'][$key])) {
+                $chunk[$paramName.'-default'] = strval($matches['escapedDefault'][$key]);
+            }
+            if (!empty($matches['unescapedDefault'][$key])) {
+                $chunk[$paramName.'-default'] = floatval($matches['unescapedDefault'][$key]);
+            }
+        }
+        return $chunk;
+    }
+
+    /**
+     * Compiles single chunk
+     * @param  {ContentBlock} $chunk     ContentBlock instance
+     * @param  {array} $arguments Arguments for this chunk from original content
+     * @return {string}            Result string ready for replacing
+     */
+    public static function compileChunk($chunk, $arguments) {
+        $matches = [];
+        $formatParams = null;
+        preg_match_all('%['.static::$prefix.']{2}([\+\*\%])([^\]\[]+)['.static::$suffix.']{2}%ui', $chunk, $matches);
+        foreach ($matches[2] as $k => $rawParam) {
+            $token = $matches[1][$k];
+            if (false === mb_strpos($rawParam, ':')) {
```

Comment:
Better use regexp here - one for retrieving param injections - second for retrieving formatter&it's attributes